File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/exception/bl/LocationPermissionDeniedException.java
package com.happs.ximand.ringcontrol.model.object.exception.bl;

import com.happs.ximand.ringcontrol.R;

public class LocationPermissionDeniedException extends BluetoothException {

    @Override
    public int getMessageResId() {
        return R.string.location_permission_denied;
    }

}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/exception/LocationPermissionDeniedException.kt
package com.happs.ximand.ringcontrol.model.`object`.exception

import com.happs.ximand.ringcontrol.R

class LocationPermissionDeniedException : BaseException(CODE) {

    companion object {
        const val CODE = 0x10
    }

    override val descriptionResId: Int = R.string.location_perm_deny_description
}